amdgpu: panic Unregistered use of FPU in kernel

**Describe the bug**
Testing on drm-kmod master and freebsd main (16-CURRENT e9fc0c538264355bd3fd9120c650078281c2a290). 

I've stuck a core in my freefall home dir (thj/amd-panic.tar.gz)

**To Reproduce**
Trigger on kldload

**Additional context**
Add any other context about the problem here.

```
amdgpu: ATOM BIOS: 113-STRIXEMU-001
drmn0: successfully loaded firmware image 'amdgpu/psp_14_0_4_toc.bin'
drmn0: successfully loaded firmware image 'amdgpu/psp_14_0_4_ta.bin'
drmn0: successfully loaded firmware image 'amdgpu/dcn_3_5_dmcub.bin'
drmn0: successfully loaded firmware image 'amdgpu/gc_11_5_2_pfp.bin'
drmn0: successfully loaded firmware image 'amdgpu/gc_11_5_2_me.bin'
drmn0: successfully loaded firmware image 'amdgpu/gc_11_5_2_rlc.bin'
drmn0: successfully loaded firmware image 'amdgpu/gc_11_5_2_mec.bin'
drmn0: successfully loaded firmware image 'amdgpu/gc_11_5_2_imu.bin'
drmn0: successfully loaded firmware image 'amdgpu/sdma_6_1_2.bin'
drmn0: successfully loaded firmware image 'amdgpu/vcn_4_0_5.bin'
drmn0: successfully loaded firmware image 'amdgpu/gc_11_5_2_mes_2.bin'
drmn0: successfully loaded firmware image 'amdgpu/gc_11_5_2_mes1.bin'
drmn0: VPE: collaborate mode falsedrmn0: Trusted Memory Zone (TMZ) feature disabled as experimental (default)
drmn0: PCIE atomic ops is not supported
[drm] vm size is 262144 GB, 4 levels, block size is 9-bit, fragment size is 9-bit
drmn0: VRAM: 512M 0x0000008000000000 - 0x000000801FFFFFFF (512M used)
drmn0: GART: 512M 0x00007FFF00000000 - 0x00007FFF1FFFFFFF
[drm] Detected VRAM RAM=512M, BAR=512M
[drm] RAM width 128bits DDR5
[drm] amdgpu: 512M of VRAM memory ready
[drm] amdgpu: 11905M of GTT memory ready.
[drm] GART: num cpu pages 131072, num gpu pages 131072
[drm] PCIE GART of 512M enabled (table at 0x0000008000F00000).
[drm] Loading DMUB firmware via PSP: version=0x09002600
[drm] Found VCN firmware Version ENC: 1.24 DEC: 9 VEP: 0 Revision: 13
drmn0: successfully loaded firmware image 'amdgpu/vpe_6_1_3.bin'
drmn0: reserve 0x1100000 from 0x801c000000 for PSP TMR
drmn0: RAS: optional ras ta ucode is not available
drmn0: RAP: optional rap ta ucode is not available
drmn0: SECUREDISPLAY: securedisplay ta ucode is not available
drmn0: SMU is initialized successfully!
[drm] Seamless boot condition check passed
panic: Unregistered use of FPU in kernel
cpuid = 10
time = 1776067725
KDB: stack backtrace:
db_trace_self_wrapper() at db_trace_self_wrapper+0x2b/frame 0xfffffe00ebee10b0
vpanic() at vpanic+0x136/frame 0xfffffe00ebee11e0
panic() at panic+0x43/frame 0xfffffe00ebee1240
trap() at trap+0xa1d/frame 0xfffffe00ebee1370
calltrap() at calltrap+0x8/frame 0xfffffe00ebee1370
--- trap 0x16, rip = 0xffffffff8436c674, rsp = 0xfffffe00ebee1448, rbp = 0xfffffe00ebee1448 ---
dml2_init_socbb_params() at dml2_init_socbb_params+0x4/frame 0xfffffe00ebee1448
dml2_init() at dml2_init+0xac/frame 0xfffffe00ebee1468
dml2_create() at dml2_create+0x63/frame 0xfffffe00ebee1490
dc_state_create() at dc_state_create+0xba/frame 0xfffffe00ebee14c0
dc_create() at dc_create+0x3e7/frame 0xfffffe00ebee1500
dm_hw_init() at dm_hw_init+0x3f8/frame 0xfffffe00ebee1710
amdgpu_device_ip_hw_init_phase2() at amdgpu_device_ip_hw_init_phase2+0x5a/frame 0xfffffe00ebee1740
amdgpu_device_ip_init() at amdgpu_device_ip_init+0x3e0/frame 0xfffffe00ebee17b0
amdgpu_device_init() at amdgpu_device_init+0x1cd2/frame 0xfffffe00ebee1860
amdgpu_driver_load_kms() at amdgpu_driver_load_kms+0x16/frame 0xfffffe00ebee1890
amdgpu_pci_probe() at amdgpu_pci_probe+0x2e5/frame 0xfffffe00ebee18e0
linux_pci_attach_device() at linux_pci_attach_device+0x53b/frame 0xfffffe00ebee1930
device_attach() at device_attach+0x45b/frame 0xfffffe00ebee1980
bus_generic_driver_added() at bus_generic_driver_added+0x90/frame 0xfffffe00ebee19a0
devclass_driver_added() at devclass_driver_added+0x29/frame 0xfffffe00ebee19d0
devclass_add_driver() at devclass_add_driver+0x138/frame 0xfffffe00ebee1a10
_linux_pci_register_driver() at _linux_pci_register_driver+0xc1/frame 0xfffffe00ebee1a40
amdgpu_evh() at amdgpu_evh+0x73/frame 0xfffffe00ebee1a50
module_register_init() at module_register_init+0xb0/frame 0xfffffe00ebee1a80
linker_load_module() at linker_load_module+0xc56/frame 0xfffffe00ebee1d80
kern_kldload() at kern_kldload+0x16e/frame 0xfffffe00ebee1dd0
sys_kldload() at sys_kldload+0x59/frame 0xfffffe00ebee1e00
amd64_syscall() at amd64_syscall+0x169/frame 0xfffffe00ebee1f30
fast_syscall_common() at fast_syscall_common+0xf8/frame 0xfffffe00ebee1f30
--- syscall (304, FreeBSD ELF64, kldload), rip = 0x3476e4ce41ca, rsp = 0x3476e346ced8, rbp = 0x3476e346d450 ---
```

I have a working uart to the laptop and I continue to get this panic with the latest master changes and some hacks from Ed

```
<6>[drm] Initialized amdgpu 3.58.0 for drmn0 on minor 0
panic: Unregistered use of FPU in kernel
cpuid = 2
time = 1776437653
KDB: stack backtrace:
db_trace_self_wrapper() at db_trace_self_wrapper+0x2b/frame 0xfffffe00ebfeecf0
vpanic() at vpanic+0x136/frame 0xfffffe00ebfeee20
panic() at panic+0x43/frame 0xfffffe00ebfeee80
trap() at trap+0xa1d/frame 0xfffffe00ebfeefb0
calltrap() at calltrap+0x8/frame 0xfffffe00ebfeefb0
--- trap 0x16, rip = 0xffffffff8436ea44, rsp = 0xfffffe00ebfef080, rbp = 0xfffffe00ebfef080 ---
populate_dml_timing_cfg_from_stream_state() at populate_dml_timing_cfg_from_stream_state+0x54/frame 0xfffffe00ebfef080
map_dc_state_into_dml_display_cfg() at map_dc_state_into_dml_display_cfg+0x63a/frame 0xfffffe00ebfef178
dml2_validate() at dml2_validate+0xe4/frame 0xfffffe00ebfef1e0
dcn35_validate_bandwidth() at dcn35_validate_bandwidth+0x2f/frame 0xfffffe00ebfef210
dc_validate_global_state() at dc_validate_global_state+0x2b2/frame 0xfffffe00ebfef280
create_validate_stream_for_sink() at create_validate_stream_for_sink+0xf28/frame 0xfffffe00ebfef4f0
amdgpu_dm_connector_mode_valid() at amdgpu_dm_connector_mode_valid+0x195/frame 0xfffffe00ebfef540
__drm_helper_update_and_validate() at __drm_helper_update_and_validate+0x1cd/frame 0xfffffe00ebfef5c0
drm_helper_probe_single_connector_modes() at drm_helper_probe_single_connector_modes+0x488/frame 0xfffffe00ebfef680
drm_client_modeset_probe() at drm_client_modeset_probe+0x277/frame 0xfffffe00ebfef7b0
__drm_fb_helper_initial_config_and_unlock() at __drm_fb_helper_initial_config_and_unlock+0x29/frame 0xfffffe00ebfef840
drm_fbdev_ttm_client_hotplug() at drm_fbdev_ttm_client_hotplug+0x74/frame 0xfffffe00ebfef870
drm_client_register() at drm_client_register+0x7e/frame 0xfffffe00ebfef890
amdgpu_pci_probe() at amdgpu_pci_probe+0x533/frame 0xfffffe00ebfef8e0
linux_pci_attach_device() at linux_pci_attach_device+0x53b/frame 0xfffffe00ebfef930
device_attach() at device_attach+0x45b/frame 0xfffffe00ebfef980
bus_generic_driver_added() at bus_generic_driver_added+0x90/frame 0xfffffe00ebfef9a0
devclass_driver_added() at devclass_driver_added+0x29/frame 0xfffffe00ebfef9d0
devclass_add_driver() at devclass_add_driver+0x138/frame 0xfffffe00ebfefa10
_linux_pci_register_driver() at _linux_pci_register_driver+0xc1/frame 0xfffffe00ebfefa40
amdgpu_evh() at amdgpu_evh+0x73/frame 0xfffffe00ebfefa50
module_register_init() at module_register_init+0xb0/frame 0xfffffe00ebfefa80
linker_load_module() at linker_load_module+0xc56/frame 0xfffffe00ebfefd80
kern_kldload() at kern_kldload+0x16e/frame 0xfffffe00ebfefdd0
sys_kldload() at sys_kldload+0x59/frame 0xfffffe00ebfefe00
amd64_syscall() at amd64_syscall+0x169/frame 0xfffffe00ebfeff30
fast_syscall_common() at fast_syscall_common+0xf8/frame 0xfffffe00ebfeff30
--- syscall (304, FreeBSD ELF64, kldload), rip = 0xc8bae2551ca, rsp = 0xc8bab290458, rbp = 0xc8bab2909d0 ---
KDB: enter: panic
[ thread pid 4313 tid 100299 ]
Stopped at      kdb_enter+0x33: movq    $0,0x15dcdd2(%rip)
db> 
```

--- a/drivers/gpu/drm/amd/display/dc/dml2/dml21/dml21_wrapper.c
+++ b/drivers/gpu/drm/amd/display/dc/dml2/dml21/dml21_wrapper.c
@@ -87,6 +87,8 @@ static void dml21_init(const struct dc *in_dc, struct dml2_context **dml_ctx, co
 	/* Store configuration options */
 	(*dml_ctx)->config = *config;
 
+	DC_FP_START();
+
 	/*Initialize SOCBB and DCNIP params */
 	dml21_initialize_soc_bb_params(&(*dml_ctx)->v21.dml_init, config, in_dc);
 	dml21_initialize_ip_params(&(*dml_ctx)->v21.dml_init, config, in_dc);
@@ -97,6 +99,8 @@ static void dml21_init(const struct dc *in_dc, struct dml2_context **dml_ctx, co
 
 	/*Initialize DML21 instance */
 	dml2_initialize_instance(&(*dml_ctx)->v21.dml_init);
+
+	DC_FP_END();
 }
 
 bool dml21_create(const struct dc *in_dc, struct dml2_context **dml_ctx, const struct dml2_configuration_options *config)

--- a/drivers/gpu/drm/amd/display/dc/dml2/dml21/dml21_wrapper.c
+++ b/drivers/gpu/drm/amd/display/dc/dml2/dml21/dml21_wrapper.c
@@ -281,11 +281,16 @@ bool dml21_validate(const struct dc *in_dc, struct dc_state *context, struct dml
 {
 	bool out = false;
 
+	DC_FP_START();
+
 	/* Use dml_validate_only for fast_validate path */
-	if (fast_validate) {
+	if (fast_validate)
 		out = dml21_check_mode_support(in_dc, context, dml_ctx);
-	} else
+	else
 		out = dml21_mode_check_and_programming(in_dc, context, dml_ctx);
+
+	DC_FP_END();
+
 	return out;
 }
 

--- a/drivers/gpu/drm/amd/display/dc/dml2/dml21/dml21_wrapper.c
+++ b/drivers/gpu/drm/amd/display/dc/dml2/dml21/dml21_wrapper.c
@@ -420,8 +420,12 @@ void dml21_copy(struct dml2_context *dst_dml_ctx,
 
 	dst_dml_ctx->v21.mode_programming.programming = dst_dml2_programming;
 
+	DC_FP_START();
+
 	/* need to initialize copied instance for internal references to be correct */
 	dml2_initialize_instance(&dst_dml_ctx->v21.dml_init);
+
+	DC_FP_END();
 }
 
 bool dml21_create_copy(struct dml2_context **dst_dml_ctx,

--- a/drivers/gpu/drm/amd/display/dc/dml2/dml2_wrapper.c
+++ b/drivers/gpu/drm/amd/display/dc/dml2/dml2_wrapper.c
@@ -719,11 +719,16 @@ bool dml2_validate(const struct dc *in_dc, struct dc_state *context, struct dml2
 		return out;
 	}
 
+	DC_FP_START();
+
 	/* Use dml_validate_only for fast_validate path */
 	if (fast_validate)
 		out = dml2_validate_only(context);
 	else
 		out = dml2_validate_and_build_resource(in_dc, context);
+
+	DC_FP_END();
+
 	return out;
 }
 

--- a/drivers/gpu/drm/amd/display/dc/dml2/dml2_wrapper.c
+++ b/drivers/gpu/drm/amd/display/dc/dml2/dml2_wrapper.c
@@ -764,11 +764,15 @@ static void dml2_init(const struct dc *in_dc, const struct dml2_configuration_op
 		break;
 	}
 
+	DC_FP_START();
+
 	initialize_dml2_ip_params(*dml2, in_dc, &(*dml2)->v20.dml_core_ctx.ip);
 
 	initialize_dml2_soc_bbox(*dml2, in_dc, &(*dml2)->v20.dml_core_ctx.soc);
 
 	initialize_dml2_soc_states(*dml2, in_dc, &(*dml2)->v20.dml_core_ctx.soc, &(*dml2)->v20.dml_core_ctx.states);
+
+	DC_FP_END();
 }
 
 bool dml2_create(const struct dc *in_dc, const struct dml2_configuration_options *config, struct dml2_context **dml2)

